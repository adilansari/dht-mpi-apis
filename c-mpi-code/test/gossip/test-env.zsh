#!/bin/zsh

# Be sure to make tests with D=1

source tools/test-helpers.zsh

# set -x

PROGRAM=$1
OUTPUT=${PROGRAM%.x}.out

crash()
{
  print $1
  mv ${OUTPUT} ${OUT}.failed
  exit 1
}

${LAUNCH} ${PROGRAM} > ${OUTPUT} 2>&1
CODE=$?
[[ ${CODE} == 0 ]] || crash "exit code was: ${CODE}"

CMPI_DEBUG_MASK=example ${LAUNCH} ${PROGRAM} >> ${OUTPUT} 2>&1
CODE=$?
[[ ${CODE} == 0 ]] || crash "exit code was: ${CODE}"

CMPI_DEBUG_MASK=all ${LAUNCH} ${PROGRAM} >> ${OUTPUT} 2>&1
CODE=$?
[[ ${CODE} == 0 ]] || crash "exit code was: ${CODE}"

# Should be 3 "test program"s
N=$( grep -c "test program" ${OUTPUT} )
(( N == 3 )) || crash "N != 3"

# Should be 2 "Example debugging message."s
N=$( grep -c "EXAMPLE MESSAGE" ${OUTPUT} )
(( N == 2 )) || crash "N != 2"

exit 0
