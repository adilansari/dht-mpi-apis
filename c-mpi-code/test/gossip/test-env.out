This test program will print an example debugging message on the final
line of output if the CMPI_DEBUG_MASK environment variable is set to include
"example" or "all" in its comma separated list of masks:
This test program will print an example debugging message on the final
line of output if the CMPI_DEBUG_MASK environment variable is set to include
"example" or "all" in its comma separated list of masks:
[D 01:10:41.574605 R:0] EXAMPLE MESSAGE
This test program will print an example debugging message on the final
line of output if the CMPI_DEBUG_MASK environment variable is set to include
"example" or "all" in its comma separated list of masks:
[D 01:10:41.583031 R:0] EXAMPLE MESSAGE
